python/src/viewer.py

```diff
@@ -21,6 +21,14 @@
 import re
 import warnings
 import io
+import argparse
+
+# Allows for hiding figures if needed from the command line
+parser = argparse.ArgumentParser(description='Viewer parser.')
+parser.add_argument("--noshow", action="store_true", help="Hide figures if used.", dest="noshow", default=False )
+args = parser.parse_args()
+_noshow = args.noshow
```

Comment:
use a non-interactive matplotlib backend (agg) instead
you can set the env variable MPLBACKEND=agg as a test property